tests/test_tdigest.py

```diff
@@ -110,8 +109,10 @@ def test_tdigest_multi_device(
         np.testing.assert_allclose(expected_median, actual_median, rtol=0.01)
 
 
-@requires_crick
-def test_load_from_checkpoint_multi_device(tmp_path: Path):
+def test_load_from_checkpoint_multi_device():
+    # using pytest tmp_path fixture creates a directory per rank
+    # so we need to use a fixed shared directory
+    tmp_path = Path("/tmp/test_load_from_checkpoint_multi_device")
```

Comment:
I am not entirely sure since I am still new to using tempfile and tempdirs, but perhaps this approach would be better

https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
The problem with `tempfile` is that it creates a random directory each time it is called. When `devices=2` pytorch lightning calls the `test_load_from_checkpoint_multi_device` script again for the second device and `tempfile` will create a different random directory.

But I like the idea of cleaning up so I added `shututil.rmtree(tmp_path)` to clean up the directory.